Library/Homebrew/brew.sh

```diff
@@ -247,7 +247,11 @@ then
   # Hide shellcheck complaint:
   # shellcheck source=/dev/null
   source "$HOMEBREW_BASH_COMMAND"
-  { update-preinstall; "homebrew-$HOMEBREW_COMMAND" "$@"; exit $?; }
```

Comment:
By the way, things will become funny in most cases if the `exit 0` is omitted:

```console
$ ./test-1
test-1: Goodbye World!
$ ./test-2
test-2: Hello World!
./test-2: line 7: syntax error near unexpected token `}'
./test-2: line 7: `}'
$ ./test-3
test-3: Hello World!
./test-3: line 4: unexpected EOF while looking for matching `"'
./test-3: line 5: syntax error: unexpected end of file
$ ./test-4
test-4: Hello World!
./test-4: line 4: syntax error near unexpected token `}'
./test-4: line 4: `}'
```

That's another important thing we rely on: Stop before leaving this group command.